src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotpyeIterator.java

```diff
@@ -0,0 +1,108 @@
+package org.broadinstitute.hellbender.tools.walkers.genotyper;
+
+import org.broadinstitute.hellbender.tools.walkers.genotyper.GLVectorSizeCache;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+/**
+ * Handles addressing general ploidy genotype likelihoods. We store likelihoods as a map
+ * of form int[] -> double (to be more precise, IntArrayWrapper -> Double).
+ * For a given ploidy (chromosome count) and number of alleles, we need a form to iterate deterministically
+ * across all possible allele conformations, that is, all possible ways in which N whole numbers will sum to P,
+ * where N is number of alleles and P is number of chromosomes.
+ * For example, with P=2,N=2, ploidy = 2 iterator will produce
+ * alleleCounts = [2 0 ] [1 1] [ 0 2]
+ *
+ *
+ */
+public final class GenotpyeIterator {
```

Comment:
This functionality is available thru ```GenotypeLikelihoodCalculator```. It does not provide an iterator per se but it caches the different allele-count combinations using ```GenotypeLikelihoodCalculator#genotypeAlleleCountsAt```.

In fact GenotypeAlleleCounts is actually capable of acting as an iterator (check on the protected increase  method). It is not public as to make GenotypeAlleleCounts mostly a constant object. You could change that but I rather you stick to use ```GenotypeLikelihoodCalculator#genotypeAlleleCountsAt```.